litmus-portal/authentication/api/handlers/dex_auth_handler.go

```diff
@@ -0,0 +1,123 @@
+package handlers
+
+import (
+	"context"
+	"litmus/litmus-portal/authentication/api/presenter"
+	"litmus/litmus-portal/authentication/pkg/entities"
+	"litmus/litmus-portal/authentication/pkg/user"
+	"litmus/litmus-portal/authentication/pkg/utils"
+	"net/http"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+)
+
+var (
+	state    = "holderState"
```

Comment:
This has the same issues as the github/google oauth. The state can be overridden by another simultaneous call to the dex, so verification will not work. Please update the state management to the same method used for google/github oauth

Reply:
Good catch! In order to manage state, we would probably have to use db or an in-memory array to maintain the same. I prefer an in-memory array, which should work.

> Please update the state management to the same method used for google/github oauth

Which method are you referring to? There is no Google/GitHub auth implementation, hence we have Dex that handles it for us.

Reply:
Okay, we did not add it to litmus. For the state solution, I suggest using jwt tokens, create a short-lived jwt token that can be verified in the callback. If the jwt token is valid then allow the login. Using jwt mean we don't need either a db or in-memory storage.

Reply:
JWT wouldn't make sense here imo, cause we will be redirecting the client to Dex Server and DexServer will be creating the callback, we cannot modify the callback from DexServer to send additional headers and pass JWT. Hence the case would fail.